### DISCUSSION TITLE
[help] Has anyone tried to call this library in C#?

### DISCUSSION THREAD

**Discussion by BruceQiu1996** [eProsima/Fast-DDS#5430] [Unofficial support]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

Our company has done some host computer programs in C#, but we need to use fastdds to communicate with some other programs。So our C++ team encapsulates a DLL and exposes the publish and subscribe interfaces for us to call.

### Current behavior

We have found that after using it for a long time or occasionally, the program crashes.I saw some memory errors in the Windows logs, probably because unsafe memory was accessed, as shown in the figure

### Steps to reproduce

Just c# call c++ dll for fastdds

### Fast DDS version/commit

2.12.1

### Platform/Architecture

Windows 10 Visual Studio 2019

### Transport layer

Default configuration, UDPv4 & SHM

### Additional context

<img width="441" alt="企业微信截图_17327038257120" src="https://github.com/user-attachments/assets/ecc80819-c662-4f76-bdb6-054afcc2fe10">

**Comment by cferreiragonz** [comment]
Hi @BruceQiu1996, thanks for using Fast DDS!

I am moving this issue to the corresponding discussion forum since `Fast DDS 2.12` is no longer a maintained version and the issue might be related with the custom C# wrapper you are using.

If you are interested in a more detailed analysis you may also consider reaching out to [commercial support](https://github.com/eProsima/Fast-DDS?tab=readme-ov-file#commercial-support) for premium support options.
